contracts/crowdfund/Crowdfund.sol

```diff
@@ -521,68 +549,71 @@ abstract contract Crowdfund is Implementation, ERC721Receiver, CrowdfundNFT {
         uint96 amount,
         address delegate,
         uint96 previousTotalContributions,
-        bytes memory gateData
+        bytes memory gateData,
+        bool skipDelegation
     ) private {
-        // Require a non-null delegate.
-        if (delegate == address(0)) {
-            revert InvalidDelegateError();
-        }
```

Comment:
maybe we should just refactor this out into a `_setDelegate()` fn that gets called by `contribute()` and `contributeFor()`? The skip logic feels awkward here which means this fn is/was probably doing too much.